shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt

```diff
@@ -5,20 +5,28 @@ import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.launch
+import org.jetbrains.compose.resources.ExperimentalResourceApi
 
-class ListScreenViewModel {
+class ListScreenViewModel : SortInteraction {
     private val viewModelScope = CoroutineScope(Dispatchers.Main)
     private val countriesApi = CountriesApiImpl()
     val state = MutableStateFlow<ListScreenState>(ListScreenState.Loading)
 
     init {
+        fetchCountries()
+    }
+
+    private fun fetchCountries(sortOrder: SortOrder = SortOrder.ASCENDING) {
         viewModelScope.launch(Dispatchers.Main) {
             try {
-                val countriesList = countriesApi.getCountriesList()
+                val countries = getCountriesSorted(
+                    countriesApi.getCountriesList(),
+                    sortOrder
+                )
                 state.emit(
                     ListScreenState.Success(
-                        countriesList = countriesList,
-                        selectedCountry = countriesList.first(),
+                        countriesList = countries,
+                        selectedCountry = countries.first()
```

Comment:
What about doing the sorting there directly like
```

val sortedCountriesList = if (isAscending) {
    countriesApi.getCountriesList().sortedBy { it.name }
} else {
    countriesApi.getCountriesList().sortedByDescending { it.name }
}
```

Reply:
@SEAbdulbasit I need to change sort by **Countries name** and **TouristPlaces name** in one function. My code handles this case.